# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/gfxFT2Utils.cpp
# Commit: 3b719cd12933
# Full Hash: 3b719cd1293370d843691b3656b52b04042d5ec4
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2022-03-02 04:31:47
# Regressor Bug: 1756720
# File Overlap Count: 1
# Description:
#   Bug 1756720 - When using legacy MS Symbol fonts, map character codes from the U+00xx range to the PUA range U+F0xx found in the cmap subtables. r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D139724
# ==============================================================================

diff -r 9849c6e79fa3 -r 3b719cd12933 gfx/thebes/gfxFT2Utils.cpp
--- a/gfx/thebes/gfxFT2Utils.cpp	Tue Mar 01 17:27:27 2022 +0200
+++ b/gfx/thebes/gfxFT2Utils.cpp	Tue Mar 01 15:22:56 2022 +0000
@@ -36,10 +36,16 @@
     }
   }
 
-  return FcFreeTypeCharIndex(mFace, aCharCode);
+  uint32_t gid = FcFreeTypeCharIndex(mFace, aCharCode);
 #else
-  return FT_Get_Char_Index(mFace, aCharCode);
+  uint32_t gid = FT_Get_Char_Index(mFace, aCharCode);
 #endif
+  if (!gid && mFace->charmap->encoding == FT_ENCODING_MS_SYMBOL) {
+    if (auto pua = gfxFontUtils::MapLegacySymbolFontCharToPUA(aCharCode)) {
+      gid = FT_Get_Char_Index(mFace, pua);
+    }
+  }
+  return gid;
 }
 
 typedef FT_UInt (*GetCharVariantFunction)(FT_Face face, FT_ULong charcode,